src/impl/util.js

```diff
@@ -88,9 +88,17 @@ export function padStart(input, n = 2) {
   }
 }
 
+export function parseInteger(string) {
+  if (isUndefined(string) || string === null || string === "") {
+    return undefined;
+  } else {
+    return parseInt(string, 10);
+  }
+}
+
 export function parseMillis(fraction) {
-  if (isUndefined(fraction)) {
-    return NaN;
+  if (isUndefined(fraction) || fraction === null || fraction === "") {
+    return undefined;
```

Comment:
Why this?

Reply:
Testing for `null`, `undefined` and "" since all three would otherwise result in a 0 output on the next line. Semantically, milliseconds are not set in these cases, so 0 is not an actual representation (although it would probably break no tests).

Using `undefined` because it will be ignored, while `NaN` would now be considered an invalid value.

I'll add a comment.